### ISSUE TITLE
sqlalchemy connection & pandas `object` dtype

### ISSUE THREAD

**Issue by dmvianna** [apple/turicreate#92]
It seems `turicreate` doesn't like `sqlalchemy` connections. Fair enough. I can connect to it using `pandas` instead and then create an `SFrame` from it.

Turns out, though, that in `pandas` string types are represented under the umbrella of `dtype(O)`. And `turicreate` doesn't recognise these as strings.

`pandas` and `sqlalchemy` play really nice with each other, and I would wage there's a lot of people in the community who use these in conjunction to simplify fast prototyping. Is this problem something you would like to look into?

**Comment by srikris** [comment]
You are right, the SFrame `from_sql` function currently does not work with SQLAlchemy. [Pandas documentation](https://pandas.pydata.org/pandas-docs/version/0.21/generated/pandas.read_sql_query.html) suggests that they support:

* sqlite3 DBAPI2
* SQLAlchemy connectable(engine/connection)
* database string URI

The SFrame [from_sql](https://github.com/apple/turicreate/blob/6225cd5e31f743d10b9cd3b5f4c6d1cca683d161/src/unity/python/turicreate/data_structures/sframe.py#L1611) function only supports DBAPI2. It would be awesome if you could look into SQLAlchemy. This wasn't on our immediate radar.

Here are some pointers that can help you with this:
*  [SFrame Builder class](https://apple.github.io/turicreate/docs/api/generated/turicreate.SFrameBuilder.html)
* [from_sql source code](https://github.com/apple/turicreate/blob/6225cd5e31f743d10b9cd3b5f4c6d1cca683d161/src/unity/python/turicreate/data_structures/sframe.py#L1611)

Feel free to ask us any questions.

**Comment by Mithilan16** [comment]
Hi! Can I take this bug?

**Comment by srikris** [comment]
@Mithilan16 Yes you may!

**Comment by znation** [comment]
@karthikdash This isn't fixed yet. Please feel free to investigate and put up a PR with a fix, thanks!
